addons/website_sale_loyalty/controllers/main.py

```diff
@@ -14,6 +14,8 @@ class WebsiteSale(main.WebsiteSale):
     @http.route()
     def pricelist(self, promo, **post):
         order = request.website.sale_get_order()
+        if not order:
+            return request.redirect(post.get('r', '/shop'))
```

Comment:
I don't think we should redirect to the post redirect parameter here.
```suggestion
        if not order:
            return request.redirect('/shop')
```